### ISSUE TITLE
Feature Request/Discussion: Adding Validation Against Enum?

### ISSUE THREAD

**Issue by Lechus** [JeremySkinner/FluentValidation#208]
Hi,

Is there a chance to have built-in Enum validator?

```
enum SomeEnums {
a =1,
b=3
}
RuleFor(x=>x.someEnum).ValidEnum<SomeEnums> ?
```

Default someEnum will have value of 0.

**Comment by waynebrantley** [comment]
I think that could be done fairly easy.  Have you tried to write it using SetValidator() and setting it to a custom validator you write?

**Comment by Lechus** [comment]
Hi @waynebrantley.
I found this topic [http://stackoverflow.com/questions/13148825/validating-enums-with-a-custom-fluentvalidator-validator](url) and it works for me.
But it would be nice to have it built-in.

**Comment by waynebrantley** [comment]
@Lechus looks like a nice feature to me.  If you do a quick pull request with this feature and a test, I am positive @JeremySkinner will accept the pull request.   I for one would use your contribution.  You up for it?

**Comment by JeremySkinner** [comment]
Sure; if you want to submit a pull request including appropriate tests I can get that merged in for you.

**Comment by Lechus** [comment]
Hi @JeremySkinner. I have made a pull request Enum validator #214

**Comment by JeremySkinner** [comment]
Thanks! I'll try and get this merged in later this week.

**Comment by JeremySkinner** [comment]
I've merged it in, and made a few tweaks.

I made the EnumValidator class itself non-generic. Typically with FluentValidation, the validator classes themselves should non-generic, which makes with easier to work with when generating metadata. These are then wrapped in the fluent interface, which provides the more user-friendly, generic syntax. 

Also, it wouldn't compile in the PCL build- reflection calls are done differently in newer versions of .NET, so it's necessary to introduce an abstraction layer that can handle these differences (type.IsEnum vs type.GetTypeInfo().IsEnum). Typically this is done with #ifdefs, which I have all in the Compatibility.cs file.
